cpp/src/arrow/tensor.cc

```diff
@@ -285,74 +288,78 @@ struct ConvertColumnsToTensorRowMajorVisitor {
 };
 
 template <typename DataType>
-inline void ConvertColumnsToTensor(const RecordBatch& batch, uint8_t* out,
-                                   bool row_major) {
+inline void ConvertColumnsToTensor(const Table& table, uint8_t* out, bool row_major) {
   using CType = typename arrow::TypeTraits<DataType>::CType;
   auto* out_values = reinterpret_cast<CType*>(out);
 
   int i = 0;
-  for (const auto& column : batch.columns()) {
-    if (row_major) {
-      ConvertColumnsToTensorRowMajorVisitor<CType> visitor{out_values, *column->data(),
-                                                           batch.num_columns(), i++};
-      DCHECK_OK(VisitTypeInline(*column->type(), &visitor));
-    } else {
-      ConvertColumnsToTensorVisitor<CType> visitor{out_values, *column->data()};
-      DCHECK_OK(VisitTypeInline(*column->type(), &visitor));
+  for (const auto& column : table.columns()) {
+    int j = 0;
+    for (const auto& chunk : column->chunks()) {
+      if (row_major) {
+        ConvertArrayToTensorRowMajorVisitor<CType> visitor{out_values, *chunk->data(),
+                                                           table.num_columns(), i, j};
```

Comment:
While here we can maybe use the more descriptive `col_idx` and `chunk_idx` (the names that are also used in `ConvertArrayToTensorRowMajorVisitor`)